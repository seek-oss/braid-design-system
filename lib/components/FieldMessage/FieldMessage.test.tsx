import React from 'react';
import { FieldMessage } from '..';
import { render } from 'react-testing-library';
import forEachTheme from '../../test/utils/forEachTheme';

describe('FieldMessage', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('False', () => {
      const { container } = render(
        <ThemeProvider>
          <FieldMessage message={false} />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Empty', () => {
      const { container } = render(
        <ThemeProvider>
          <FieldMessage message="" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Default', () => {
      const { container } = render(
        <ThemeProvider>
          <FieldMessage message="Message" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Critical', () => {
      const { container } = render(
        <ThemeProvider>
          <FieldMessage message="Message" tone="critical" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Positive', () => {
      const { container } = render(
        <ThemeProvider>
          <FieldMessage message="Message" tone="positive" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
