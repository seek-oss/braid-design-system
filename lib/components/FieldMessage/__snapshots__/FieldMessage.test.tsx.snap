// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldMessage Theme: jobStreet Critical 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__critical fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__critical Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 512 512"
          width="16"
        >
          <path
            d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: jobStreet Default 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: jobStreet Empty 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div />
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: jobStreet False 1`] = `null`;

exports[`FieldMessage Theme: jobStreet Positive 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__positive fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__positive Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAnz Critical 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__critical fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__critical Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 512 512"
          width="16"
        >
          <path
            d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAnz Default 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAnz Empty 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div />
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAnz False 1`] = `null`;

exports[`FieldMessage Theme: seekAnz Positive 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__positive fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__positive Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAsia Critical 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__critical fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__critical Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 512 512"
          width="16"
        >
          <path
            d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAsia Default 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAsia Empty 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div />
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: seekAsia False 1`] = `null`;

exports[`FieldMessage Theme: seekAsia Positive 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__positive fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__positive Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: wireframe Critical 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__critical fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__critical Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 512 512"
          width="16"
        >
          <path
            d="M489.308 156.341c-12.902-30.484-31.355-57.856-54.861-81.357-23.512-23.501-50.883-41.959-81.377-54.855-31.575-13.353-65.106-20.122-99.65-20.122s-68.066 6.769-99.635 20.122c-30.484 12.902-57.856 31.35-81.357 54.851-23.511 23.501-41.968 50.877-54.865 81.357-13.342 31.575-20.122 65.111-20.122 99.656 0 34.56 6.774 68.080 20.122 99.656 12.902 30.485 31.36 57.856 54.861 81.367s50.877 41.959 81.357 54.856c31.575 13.348 65.096 20.122 99.635 20.122 34.56 0 68.081-6.759 99.656-20.122 30.485-12.902 57.856-31.35 81.372-54.856 23.5-23.5 41.964-50.883 54.861-81.367 13.368-31.58 20.137-65.106 20.137-99.661 0-34.544-6.774-68.076-20.132-99.645zM232.99 371.167h41.334c3.687 0 6.687 2.995 6.687 6.692v41.063c0 3.702-2.995 6.697-6.692 6.697h-41.339c-3.696 0-6.692-2.995-6.692-6.692v-41.063c0-3.702 2.99-6.697 6.686-6.697zM226.304 312.051v-219.587c0-3.697 2.99-6.692 6.686-6.692h41.063c3.707 0 6.707 3 6.707 6.697v219.587c0 3.697-3 6.692-6.707 6.692h-41.063c-3.696 0-6.692-3-6.692-6.692z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: wireframe Default 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: wireframe Empty 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__neutral fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div />
    </div>
  </div>
</div>
`;

exports[`FieldMessage Theme: wireframe False 1`] = `null`;

exports[`FieldMessage Theme: wireframe Positive 1`] = `
<div
  class="Box__root borderColor__standard paddingBottom__small"
  tabindex="-1"
>
  <div
    class="Text__block fontFamily__text color__positive fontSize__standard fontWeight__regular transform__standardText Box__root borderColor__standard"
  >
    <div
      class="FieldMessage__content"
    >
      <div
        class="height__standardText"
      />
      <div
        class="FieldMessage__icon Box__root borderColor__standard paddingRight__xsmall"
      >
        <svg
          class="width__standardText height__standardText fill__positive Icon__block Box__root borderColor__standard"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
          />
        </svg>
      </div>
      <div>
        Message
      </div>
    </div>
  </div>
</div>
`;
